compiler/src/dotty/tools/backend/sjs/JUnitBootstrappers.scala

```diff
@@ -0,0 +1,341 @@
+    val sym = ctx.newSymbol(owner, name, Synthetic | Method,
+        MethodType(Names.instance :: Nil, defn.ObjectType :: Nil, defn.UnitType)).entered
+
+    DefDef(sym, { (paramRefss: List[List[Tree]]) =>
+      val List(List(instanceParamRef)) = paramRefss
+      val calls = annotatedMethods(testClass, annot)
+        .map(m => Apply(instanceParamRef.cast(testClass.typeRef).select(m), Nil))
+      Block(calls, unitLiteral)
+    })
+  }
+
+  private def genTests(owner: ClassSymbol, tests: List[Symbol])(
+      implicit ctx: Context, mydefn: MyDefinitions): DefDef = {
+
+    val sym = ctx.newSymbol(owner, Names.tests, Synthetic | Method,
+        MethodType(Nil, defn.ArrayOf(mydefn.TestMetadataType))).entered
+
+    DefDef(sym, {
+      val metadata = for (test <- tests) yield {
+        val name = Literal(Constant(test.name.toString))
+        val ignored = Literal(Constant(test.hasAnnotation(mydefn.IgnoreAnnotClass)))
+        // TODO Handle @Test annotations with arguments
+        // val reifiedAnnot = New(mydefn.TestAnnotType, test.getAnnotation(mydefn.TestAnnotClass).get.arguments)
+        val testAnnot = test.getAnnotation(mydefn.TestAnnotClass).get
+        if (testAnnot.arguments.nonEmpty)
+          ctx.error("@Test annotations with arguments are not yet supported in Scala.js for dotty", testAnnot.tree.sourcePos)
+        val noArgConstr = mydefn.TestAnnotType.member(nme.CONSTRUCTOR).suchThat(_.info.paramInfoss.head.isEmpty).symbol.asTerm
+        val reifiedAnnot = New(mydefn.TestAnnotType, noArgConstr, Nil)
+        New(mydefn.TestMetadataType, List(name, ignored, reifiedAnnot))
+      }
+      JavaSeqLiteral(metadata, TypeTree(mydefn.TestMetadataType))
+    })
+  }
+
+  private def genInvokeTest(owner: ClassSymbol, testClass: ClassSymbol, tests: List[Symbol])(
+      implicit ctx: Context, mydefn: MyDefinitions): DefDef = {
+
+    val sym = ctx.newSymbol(owner, Names.invokeTest, Synthetic | Method,
+        MethodType(List(Names.instance, Names.name), List(defn.ObjectType, defn.StringType), mydefn.FutureType)).entered
+
+    DefDef(sym, { (paramRefss: List[List[Tree]]) =>
+      val List(List(instanceParamRef, nameParamRef)) = paramRefss
+      val castInstanceSym = ctx.newSymbol(sym, Names.castInstance, Synthetic, testClass.typeRef, coord = owner.span)
+      Block(
+        ValDef(castInstanceSym, instanceParamRef.cast(testClass.typeRef)) :: Nil,
+        tests.foldRight[Tree] {
+          val tp = mydefn.NoSuchMethodExceptionType
+          val constr = tp.member(nme.CONSTRUCTOR).suchThat { c =>
+            c.info.paramInfoss.head.size == 1 &&
+            c.info.paramInfoss.head.head.isRef(defn.StringClass)
+          }.symbol.asTerm
+          Throw(New(tp, constr, nameParamRef :: Nil))
+        } { (test, next) =>
+          If(Literal(Constant(test.name.toString)).select(defn.Any_equals).appliedTo(nameParamRef),
+              genTestInvocation(testClass, test, ref(castInstanceSym)),
```

Comment:
Why four spaces for indentation here ?

Reply:
It's a continuation line. There's the same at `val sym =` in this method, for example.

I can turn all the 4-space indents into 2-space indents if you want. I'm not sure what the indent style is in this repo.

Reply:
I think we use 2-space indents for continuation, maybe we should change that convention but I'd rather avoid inconsistency if possible yes.

Reply:
However, if it makes things easier for you to keep the same indentation style here and in scala-js, then I think we can tolerate it.